_spinnaker/automated_rollbacks.md

```diff
@@ -0,0 +1,147 @@
+
+### Creating a main deployment pipeline
+
+Let’s start by creating a new pipeline that will deploy a simple nginx container:
+
+![image](/assets/images/aut-rollbacks-deploy-pipeline.png)
+
+We’ll be supplying the manifest directly in the Deploy (Manifest) stage as text for simplicity, but it can also be taken from a version control system like Github:
+
+![image](/assets/images/aut-rollbacks-deploy-manifest-1.png)
+
+This is the sample deployment manifest, which will create two nginx pods in kubernetes under the “german-env1” namespace (the namespace must exist already):
+
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: nginx-deployment
+  namespace: german-env1
+spec:
+  selector:
+    matchLabels:
+      app: nginx
+  replicas: 2
+  template:
+    metadata:
+      labels:
+        app: nginx
+    spec:
+      containers:
+      - name: nginx
+        image: nginx:1.14.0
+        ports:
+        - containerPort: 80
+```
+
+This is all that is required in the deploy pipeline. We can now save and execute manually the pipeline and it should be successful. 
+
+### Generating deployment historic versions for testing rollbacks
+
+We’ll test rollbacks by making a change to the deployment manifest and running again the pipeline, so that we’ll end up with two historic versions of the deployment. First we’ll edit the manifest to change nginx image version from `nginx:1.14.0` to `nginx:1.15.05-alpine` and then run again the pipeline. Now under infrastructure view we’ll see that we have two historic versions of the deployment:
+
+![image](/assets/images/aut-rollbacks-infra-1.png)
+
+The first version deployed (`V001`) corresponds to nginx 1.14.0, while the second version (`V002`) corresponds to nginx 1.15.5-alpine.
+
+### Creating a rollback pipeline
+
+Then we’ll be creating a rollback pipeline that triggers automatically when `deploy-nginx` pipeline fails:
+
+![image](/assets/images/aut-rollbacks-rollpipeline-1.png)
+
+![image](/assets/images/aut-rollbacks-trigger.png)
+
+This pipeline just contains a single stage of type Undo Rollout (Manifest) with the following configuration:
+
+![image](/assets/images/aut-rollbacks-undo-rollout-1.png)
+
+This configuration is important, the fields `Namespace`, `Kind` and `Name` are the coordinates of the kubernetes deployment, and should match what is defined in the deployment manifest on the deploy pipeline. `Revisions Back` indicate how many deployment versions to rollback. In our historic running we currently have `V001` and `V002`. Each time a deployment occurs, this version is automatically increased. [Click here for detailed information on how Undo Rollout (Manifest) stage works](https://www.spinnaker.io/guides/user/kubernetes-v2/automated-rollbacks/#automated-rollbacks).
+
+### Testing rollback logic
+
+Now we’ll do a rollback test by changing the manifest providing an invalid value for the namespace to force a failure. In deploy-nginx we’ll be changing the namespace to an invalid value:
```

Comment:
Add a comma after "changing the manifest".  Change "we'll be changing" to "we'll change"